Src/BootCamp.Chapter/CaesarCipher.cs

```diff
@@ -1,22 +1,242 @@
+
+            var characterOccurrences = AnalyseFrequency(encryptedMessage);
+            var possibleKeys = FindPossibleKeys(characterOccurrences);
+
+            Console.WriteLine($"{possibleKeys.Length} possible key(s) found.{Environment.NewLine}");
+            Console.WriteLine($"Adjust the bellow settings for better results!");
+            Console.WriteLine($"KeyAccuracy: {KeyAccuracy}");
+            Console.WriteLine($"Top Character: {TopCharacter}");
+            Console.WriteLine($"BaseCharacter: {BaseCharacter}");
+            Console.WriteLine();
+
+            Console.WriteLine($"Decryption results: ");
+            foreach (var key in possibleKeys)
+            {
+                var crackedMessage = Decrypt(encryptedMessage, key);
+
+                Console.WriteLine($"Key = {key} - {crackedMessage}");
+                Console.WriteLine();
+            }
+        }
+
+        /// <summary>
+        /// Find character repetition frequency in a string.
+        /// </summary>
+        /// <param name="encryptedMessage"></param>
+        /// <returns>Bi-dimensional array containing found characters and number of repetitions</returns>
+        private static int[][] AnalyseFrequency(string encryptedMessage)
+        {
+            if (!Utils.IsStringValid(encryptedMessage))
+            {
+                Console.WriteLine("Input is not a valid string");
+            }
+
+            int[] characters = new int[char.MaxValue];
+
+            foreach (char item in encryptedMessage)
+            {
+                characters[item]++;
+            }
+
+            int[] repeatedCharacters = Array.Empty<int>();
+            int[] numberOfOccurrences = Array.Empty<int>();
+
+            for (int i = 0; i < char.MaxValue; i++)
+            {
+                if (characters[i] > 0 && IsPrintableChar((char)i))
+                {
+                    repeatedCharacters = ArrayOps.InsertLast(repeatedCharacters, i);
+                    numberOfOccurrences = ArrayOps.InsertLast(numberOfOccurrences, characters[i]);
+                }
+            }
+            var analysisResult = Construct2DArray(repeatedCharacters, numberOfOccurrences);
+            return analysisResult;
+        }
+
+        /// <summary>
+        /// Transfers the contents of two one dimension arrays to one bi-dimensional array.
+        /// </summary>
+        /// <param name="array1">First int one dimension input array.</param>
+        /// <param name="array2">Second int one dimension input array.</param>
+        /// <returns>A new Bi-dimensional int array created from array1 and array2.</returns>
+        private static int[][] Construct2DArray(int[] array1, int[] array2)
+        {
+            if (ArrayOps.IsNullOrEmpty(array1) || ArrayOps.IsNullOrEmpty(array2))
+            {
+                Console.WriteLine("Input arrays are not valid!");
+                return default;
+            }
+
+            int[][] newBiDiArr = new int[][] { array1, array2 };
```

Comment:
I would think this line already "copies" (same array though) in the data. Do you really need to do the stuff after that? I did not know this. However, I rarely use those type of arrays. It's fine to not know it.

What you should ask yourself:
Can I make this work for array1, ... , arrayN? How would that general version look like?